pollux/sql-doobie/src/main/scala/org/hyperledger/identus/pollux/sql/repository/JdbcCredentialSchemaRepository.scala

```diff
@@ -17,57 +17,64 @@ import java.util.UUID
 case class JdbcCredentialSchemaRepository(xa: Transactor[ContextAwareTask], xb: Transactor[Task])
     extends CredentialSchemaRepository {
   import CredentialSchemaSql.*
-  override def create(cs: CredentialSchema): RIO[WalletAccessContext, CredentialSchema] = {
+  override def create(cs: CredentialSchema): URIO[WalletAccessContext, CredentialSchema] = {
     ZIO.serviceWithZIO[WalletAccessContext](ctx =>
       CredentialSchemaSql
         .insert(CredentialSchemaRow.fromModel(cs, ctx.walletId))
         .transactWallet(xa)
+        .orDie
         .map(CredentialSchemaRow.toModel)
     )
   }
 
-  override def getByGuid(guid: UUID): Task[Option[CredentialSchema]] = {
+  override def findByGuid(guid: UUID): UIO[Option[CredentialSchema]] = {
     CredentialSchemaSql
       .findByGUID(guid)
       .transact(xb)
+      .orDie
       .map(
         _.headOption
           .map(CredentialSchemaRow.toModel)
       )
   }
 
-  override def update(cs: CredentialSchema): RIO[WalletAccessContext, Option[CredentialSchema]] = {
+  override def update(cs: CredentialSchema): URIO[WalletAccessContext, Option[CredentialSchema]] = {
     ZIO.serviceWithZIO[WalletAccessContext](ctx =>
       CredentialSchemaSql
         .update(CredentialSchemaRow.fromModel(cs, ctx.walletId))
         .transactWallet(xa)
+        .orDie
         .map(Option.apply)
         .map(_.map(CredentialSchemaRow.toModel))
     )
   }
 
-  def getAllVersions(id: UUID, author: String): RIO[WalletAccessContext, Seq[String]] = {
+  def getAllVersions(id: UUID, author: String): URIO[WalletAccessContext, List[String]] = {
     CredentialSchemaSql
       .getAllVersions(id, author)
       .transactWallet(xa)
+      .orDie
   }
 
-  override def delete(guid: UUID): RIO[WalletAccessContext, Option[CredentialSchema]] = {
+  override def delete(guid: UUID): URIO[WalletAccessContext, Option[CredentialSchema]] = {
```

Comment:
We should "defect" when this method is called with an invalid `guid` and not return an `Option`. It should be the responsibility of the service making the call to first check that the record exists using the `find` method. See the following ADR rules:
- https://staging-docs.atalaprism.io/adrs/adr/20240116-use-zio-failures-and-defects-effectively/#reporting-404-not-found-to-user
- https://staging-docs.atalaprism.io/adrs/adr/20240116-use-zio-failures-and-defects-effectively/#do-not-return-the-affected-row-count

Reply:
fixed